src/featuredgraph.jl

```diff
@@ -0,0 +1,385 @@
+#===================================
+Define FeaturedGraph type as a subtype of LightGraphs' AbstractGraph.
+For the core methods to be implemented by any AbstractGraph, see
+https://juliagraphs.org/LightGraphs.jl/latest/types/#AbstractGraph-Type
+https://juliagraphs.org/LightGraphs.jl/latest/developing/#Developing-Alternate-Graph-Types
+=============================================#
+
+abstract type AbstractFeaturedGraph <: AbstractGraph{Int} end
+
+"""
+    NullGraph()
+
+Null object for `FeaturedGraph`.
+"""
+struct NullGraph <: AbstractFeaturedGraph end
+
+const COO_T = Tuple{T, T} where T <: AbstractVector
```

Comment:
Oh! Now I notice it. For a graph, we should support weighted graph which is the graph with weights on edges. Laplcain matrix also support weight graphs. So, the COO format should add a value for edge weight. I think it should be the same for adjacency list.

Reply:
done